cabal-testsuite/src/Test/Cabal/Prelude.hs

```diff
@@ -1136,11 +1137,8 @@ isTestFile f =
 -- function creates a directory immediately under the current drive on Windows.
 -- The directory must be passed to new- commands with --store-dir.
 withShorterPathForNewBuildStore :: (FilePath -> IO a) -> IO a
-withShorterPathForNewBuildStore test = do
-  tempDir <- if buildOS == Windows
-             then takeDrive `fmap` getCurrentDirectory
-             else getTemporaryDirectory
-  withTempDirectory normal tempDir "cabal-test-store" test
+withShorterPathForNewBuildStore test =
+  withTestDir normal "cabal-test-store" test
```

Comment:
@gbaz this implements what's requested in https://github.com/haskell/cabal/issues/7549. There's a slight infelicity to the original in that `withTestDir` doesn't accept the "current directory" where it's supposed to create the temp directory. Instead it uses `getTemporaryDirectory`. In the original version of this snippet (you should see the diff) we try to be smart and use `getTemporaryDirectory` only on non-Windows... 

The overall effect is that in the new version we'll use `getTemporaryDirectory` on all platforms. On Windows this function has a little junky logic. So I'm not sure if this is a good thing. Alternatively, I'd need to create a version of `withTestDir` that accepts the directory. Or directly modify  `withTestDir` to do the same as in the original version here (``if buildOS == Windows then takeDrive `fmap`...``). Any suggestions?

Reply:
My gut says this is just a testing function and leaving it as is should probably be ok, if our tests continue to work. In particular the generic temporary dir in windows doesn't have too long a path length, and a more general fix for this should have been implemented in https://github.com/haskell/cabal/pull/4977 as well as more recent ghcs.

Reply:
Good, thanks! Waiting for CI now...